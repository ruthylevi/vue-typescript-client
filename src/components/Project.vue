<template>
  <v-row justify="end">
    <v-dialog v-model="dialog" fullscreen hide-overlay transition="dialog-bottom-transition">
      <template v-slot:activator="{ on }">
        <v-btn color="primary" dark v-on="on">Read More</v-btn>
      </template>
      <v-card>
        <v-toolbar @click="dialog = false" dark color="primary">
          <v-btn icon dark @click="dialog = false">
            <v-icon>mdi-close</v-icon>
          </v-btn>
          <v-toolbar-title style=" font-family: Montserrat, font-size: 30px" v-text="title"></v-toolbar-title>
          <div class="flex-grow-1"></div>
        </v-toolbar>
        <v-item-group>
          <v-list-item>
            <v-list-item-content>
              <v-img :src="img" height="350"></v-img>
              <v-list-item-title class="headline text-left">Advisors</v-list-item-title>
              <v-list-item class="text-left" v-if="advisors != null" v-text="advisors.toString()"></v-list-item>
              <v-list-item-title class="headline text-left">Members</v-list-item-title>
              <v-list-item class="text-left" v-if="members != null" v-text="members.toString()"></v-list-item>
              <v-list-item-title class="headline text-left">Description</v-list-item-title>
              <v-list-item class="text-left" v-text="desc"></v-list-item>
              <v-list-item-title class="headline text-left">Contact</v-list-item-title>
              <v-list-item class="text-left" v-text="contact"></v-list-item>
              <v-list-item-title class="headline text-left">Tags</v-list-item-title>
              <v-chip-group column>
                <v-chip label v-for="tag in tags" :key="tag" class="noClick">{{tag}}</v-chip>
              </v-chip-group>
            </v-list-item-content>
          </v-list-item>

          <v-card-actions>
            <Project
              v-bind:title="title"
              v-bind:avatar="avatar"
              v-bind:img="img"
              v-bind:advisors="advisors"
              v-bind:desc="desc"
              v-bind:starred="starred"
              v-bind:tags="tags"
              v-bind:members="members"
              v-bind:contact="contact"
            ></Project>
          </v-card-actions>
        </v-item-group>
      </v-card>
    </v-dialog>
  </v-row>
</template> 

<script>
export default {
  props: {
    id: String,
    title: String,
    avatar: String,
    img: String,
    advisors: Array,
    members: Array,
    desc: String,
    starred: Boolean,
    tags: Array,
    contact: String
  },
  data() {
    return {
      dialog: false
    };
  }
};
</script>