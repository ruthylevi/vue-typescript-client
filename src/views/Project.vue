<template>
  <v-container>
    <v-row>
      <v-col cols="12" sm="8">
        <ProjectMain
          v-bind:avatar="items.thumbnail_link"
          v-bind:title="items.title"
          v-bind:starred="items.starred"
          v-bind:coverImg="items.cover_link"
          v-bind:tagline="items.tagline"
          v-bind:description="items.body"
          v-bind:majors="items.requestedMajors"
          v-bind:tags="items.tags"
          v-bind:acceptingApps="items.acceptingApplications"
          v-bind:onProject="onProject"
          v-bind:isAdmin="isAdmin"
        />
        <p />
        <ProjectBoard v-bind:boardItems="boardItems" />
      </v-col>
      <v-col cols="12" sm="4">
        <Apply
          v-bind:title="items.title"
          v-bind:acceptingApps="items.acceptingApplications"
          v-bind:onProject="onProject"
        />
        <p />
        <ContactInfo
          v-bind:title="items.title"
          v-bind:members="members"
          v-bind:advisors="advisors"
          v-bind:administrators="items.administrators"
        />
        <p />
        <Comments @comment="updateComments" v-bind:comments="comments" />
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
import apiCall from '@/apiCall';
import ProjectBoard from '@/components/ProjectBoard.vue';
import ProjectMain from '@/components/ProjectMain.vue';
import ContactInfo from '@/components/ContactInfo.vue';
import Comments from '@/components/Comments.vue';
import Apply from '@/components/Apply.vue';

export default {
  components: {
    ProjectMain,
    ProjectBoard,
    ContactInfo,
    Comments,
    Apply,
  },
  data() {
    return {
      items: [],
      comments: [
        {
          author: 'Julia Cahn',
          text: 'Example question here?',
          datetime: 'Nov 13, 2019 12:00 PM',
        },
        {
          author: 'Ruthy Levi',
          text: 'What it do bby',
          datetime: 'Nov 13, 2019 1:00 PM',
        },
        {
          author: 'Herb Zieger',
          text: 'Wow this demo looks awesome!',
          datetime: 'Nov 13, 2019 12:00 PM',
        },
        {
          author: 'Mark Freeman',
          text: 'Ooga Booga.',
          datetime: 'Nov 13, 2019 12:00 PM',
        },
        {
          author: 'Jason Chlus',
          text:
            'Yo guys! Make sure to check out my vlog channel! https://www.youtube.com/channel/UC5UiolKpoLn920GnwidWQDQ',
          datetime: 'Nov 13, 2019 12:00 PM',
        },
        {
          author: 'Jakub Kolasinski',
          text: 'A bad day on the slopes beats a good day at work!',
          datetime: 'Nov 13, 2019 12:00 PM',
        },
        {
          author: 'Tyler Lowe',
          text: 'Have any of you guys seen Jarvis?',
          datetime: 'Nov 13, 2019 12:00 PM',
        },
      ],
      boardItems: [
        {
          date: 'November 2019',
          desc: 'Actually gettting the project board done',
          img: [],
        },
        {
          date: 'October 2019',
          desc:
            'Simple UI generation from client in response to a live call from the server. Initial UI mockups for web client finished and handed over to SDEs.',
          img: [],
        },
        {
          date: 'September 2019',
          desc:
            'Schemas validated and initial routes stubbed for future implementation. Requirements validation with EMs and generation of an ordered product backlog. Developer middlewares and other tooling written to facilitate further development. Server capable of being stood up and responding to simple requests with valid data fetched from the cloud.',
        },
      ],
    };
  },
  mounted() {
    this.getProjectData();
  },
  methods: {
    async getProjectData() {
      const response = await apiCall.methods.get(
        `/projects/${this.projectId}`,
        '',
        this.$route.fullPath,
      );
      if (response.status === 200) {
        this.items = response.data;
      }
    },
    updateComments(variable) {
      const newComment = {
        author: 'Test User',
        text: variable,
        datetime: 'Nov 5, 2019 2:45 PM',
      };
      // console.log(this.comments);
      // console.log(newComment);
      this.comments = this.comments.concat(newComment);
      // console.log(this.comments);
    },
  },
  watch: {
    projectId() {
      this.getProjectData();
    },
  },
  computed: {
    projectId() {
      return this.$route.params.id;
    },
    members() {
      if (this.items.administrators === undefined || this.items.administrators === undefined) {
        return [];
      }
      return this.items.administrators
        .filter(admin => !admin.isAdvisor)
        .concat(
          this.items.contributors.filter(
            contrib => !this.items.administrators.map(admin => admin.id).includes(contrib.id),
          ),
        );
    },
    advisors() {
      if (this.items.administrators === undefined) return [];
      return this.items.administrators.filter(admin => admin.isAdvisor);
    },
    onProject() {
      return this.isAdmin || this.isContributor;
    },
    isAdmin() {
      return this.$store.getters.isAdmin(this.projectId);
    },
    isContributor() {
      return this.$store.getters.isContributor(this.projectId);
    },
  },
};
</script>
