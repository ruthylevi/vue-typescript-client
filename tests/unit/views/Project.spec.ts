import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import { shallowMount } from '@vue/test-utils';
import Project from '@/views/Project.vue';

Vue.use(Vuetify);
Vue.use(VueRouter);

const methodStub = jest.fn();

const wrapper = shallowMount(Project, {
  stubs: ['router-link', 'VChipGroup'],
  methods: {
    getProjectData: methodStub,
  },
});

describe('Project', () => {
  test('Sanity', () => {
    expect(true).toBe(true);
  });

  test('Visible', () => {
    expect(wrapper.isVisible()).toBe(true);
  });

  test('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
