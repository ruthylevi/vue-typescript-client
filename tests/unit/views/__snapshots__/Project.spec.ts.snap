// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Renders correctly 1`] = `
<v-container-stub tag="div">
  <v-row-stub tag="div">
    <v-col-stub cols="12" sm="8" tag="div">
      <projectmain-stub avatar="https://images.theconversation.com/files/93616/original/image-20150902-6700-t2axrz.jpg" coverimg="https://cdn.vox-cdn.com/uploads/chorus_image/image/47070706/google2.0.0.jpg"></projectmain-stub>
      <p></p>
      <projectboard-stub boarditems="[object Object],[object Object],[object Object]"></projectboard-stub>
    </v-col-stub>
    <v-col-stub cols="12" sm="4" tag="div">
      <v-card-stub loaderheight="4" tag="div">
        <apply-stub></apply-stub>
      </v-card-stub>
      <p></p>
      <contactinfo-stub></contactinfo-stub>
      <p></p>
      <comments-stub comments="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"></comments-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
